zerver/webhooks/front/tests.py

```diff
@@ -8,77 +8,14 @@ class FrontHookTests(WebhookTestCase):
-        payload = self.get_body(fixture_name)
-        payload_json = ujson.loads(payload)
-        del payload_json['conversation']['subject']
-        result = self.client_post(self.url, ujson.dumps(payload_json),
-                                  content_type="application/x-www-form-urlencoded")
-
-        self.assert_json_error(result, "Missing required data")
-
-    def _test_no_source_name(self, fixture_name: Text) -> None:
-        payload = self.get_body(fixture_name)
-        payload_json = ujson.loads(payload)
-        del payload_json['source']['data']['first_name']
-        result = self.client_post(self.url, ujson.dumps(payload_json),
-                                  content_type="application/x-www-form-urlencoded")
-
-        self.assert_json_error(result, "Missing required data")
-
-    def _test_no_target_name(self, fixture_name: Text) -> None:
-        payload = self.get_body(fixture_name)
-        payload_json = ujson.loads(payload)
-        del payload_json['target']['data']['first_name']
-        result = self.client_post(self.url, ujson.dumps(payload_json),
-                                  content_type="application/x-www-form-urlencoded")
-
-        self.assert_json_error(result, "Missing required data")
-
-    def _test_no_comment(self, fixture_name: Text) -> None:
-        payload = self.get_body(fixture_name)
-        payload_json = ujson.loads(payload)
-        del payload_json['target']['data']['body']
-        result = self.client_post(self.url, ujson.dumps(payload_json),
-                                  content_type="application/x-www-form-urlencoded")
-
-        self.assert_json_error(result, "Missing required data")
-
-    def _test_no_tag(self, fixture_name: Text) -> None:
-        payload = self.get_body(fixture_name)
-        payload_json = ujson.loads(payload)
-        del payload_json['target']['data']['name']
-        result = self.client_post(self.url, ujson.dumps(payload_json),
-                                  content_type="application/x-www-form-urlencoded")
-
-        self.assert_json_error(result, "Missing required data")
-
-    def test_no_event_type(self) -> None:
-        payload = self.get_body('1_conversation_assigned_outbound')
-        payload_json = ujson.loads(payload)
-        del payload_json['type']
-        result = self.client_post(self.url, ujson.dumps(payload_json),
-                                  content_type="application/x-www-form-urlencoded")
-
-        self.assert_json_error(result, "Missing required data")
-
-    def test_no_conversation_id(self) -> None:
-        payload = self.get_body('1_conversation_assigned_outbound')
-        payload_json = ujson.loads(payload)
-        del payload_json['conversation']['id']
-        result = self.client_post(self.url, ujson.dumps(payload_json),
-                                  content_type="application/x-www-form-urlencoded")
-
-        self.assert_json_error(result, "Missing required data")
-
```

Comment:
Okay, so I am a bit confused here. I see that you didn't add these tests again, what were you trying to test here? Are there scenarios where the payloads don't have the requisite data or something?

Reply:
Yep, that's exactly right. Since some of the error handling code was deleted, these tests need to be deleted as well. But to be fair, these scenarios are highly unlikely.

Reply:
Yeah, that's what I thought! :) Usually you do not have to worry about missing data when it comes to webhooks! Thanks for answering my question! :)